torchok/data/datasets/classification/multihead.py

```diff
@@ -0,0 +1,134 @@
+            annotation_path: Path to the .pkl or .csv file with path to images and annotations.
+                Path to images must be under column ``input_column`` and
+                annotations must be under ``target_column`` column.
+            targets: Dict containing information about heads
+
+                - `name` (str): Name of the output target.
+                - `column` (str): Column name containing image label.
+                - `type` (str): Format of data processing. Available values: multiclass, multilabel, feature_label.
+                - `num_classes` (int): Number of classes.
+                - `dtype` (str): Data type of the torch tensors related to the target.
+
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations`_ library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations`_ library.
+            input_column: column name containing paths to the images.
+            input_dtype: Data type of the torch tensors related to the image.
+            image_format: format of images that will be returned from dataset. Can be `rgb`, `bgr`, `rgba`, `gray`.
+            rgba_layout_color: color of the background during conversion from `rgba`.
+            test_mode: If True, only image without labels will be returned.
+            lazy_init: If True, for multilabel the target variable is converted to multihot when __getitem__ is called.
+                For multiclass will check the class index to fit the range when ``__getitem__`` is called.
+        .. _albumentations: https://albumentations.ai/docs/
+        """
+        super().__init__(
+            transform=transform,
+            augment=augment,
+            input_dtype=input_dtype,
+            image_format=image_format,
+            rgba_layout_color=rgba_layout_color,
+            test_mode=test_mode
+        )
+
+        self.data_folder = Path(data_folder)
+        self.input_column = input_column
+        self.annotation_path = annotation_path
+        self.lazy_init = lazy_init
+
+        if annotation_path.endswith('.csv'):
+            self.df = pd.read_csv(self.data_folder / annotation_path)
+        elif annotation_path.endswith('.pkl'):
+            self.df = pd.read_pickle(self.data_folder / annotation_path)
+        else:
+            raise ValueError('Detection dataset error. Annotation path is not in `csv` or `pkl` format')
+
+        self.heads = []
+        if not self.test_mode:
+            for target in targets:
+                name = target['name']
+                column = target['column']
+                target_type = target['type']
+                num_classes = target.get('num_classes', None)
+                target_dtype = target.get('dtype', 'long')
+
+                if target_type in range(len(self.TARGET_TYPES)):
```

Comment:
Why you support int value in `target_type`? In your doc it can only be string

Reply:
Removed